indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,244 @@
+        $num_examples: Int!
+    )
+    {
+        questionnaires(questionnaireIds: [$questionaire_id]) {
+            questionnaires {
+                examples(numExamples: $num_examples) {
+                    rowIndex
+                    datafileId
+                    source
+                }
+            }
+        }
+    }
+    """
+
+    def __init__(self, questionaire_id: int, num_examples: int):
+        super().__init__(
+            query=self.query,
+            variables={
+                "questionaire_id": questionaire_id,
+                "num_examples": num_examples,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            examples = [
+                Example(**e)
+                for e in super().process_response(response)["questionnaires"][
+                    "questionnaires"
+                ][0]["examples"]
+            ]
+        except IndexError:
+            raise IndicoNotFound(
+                "Examples not found. Please check the ID you are using."
+            )
+        return examples
+
+
+class CreateQuestionaire(GraphQLRequest):
+    """
+    Creates the questionnaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        dataset_id (int): The id of the dataset to create the questionnaire from.
+        source_column_id (int): The id of the source column to create a questionnaire from.
+        targets (List[str]): The classes for the dataset.
+
+    Returns:
+        Questionnaire
+
+    """
+
+    query = """
+        mutation(
+            $name: String!,
+            $dataset_id: Int!,
+            $questions: [QuestionInput]!,
+            $source_col_id: Int!,
+        ) {
+            createQuestionnaire (
+                datasetId: $dataset_id,
+                dataType: TEXT,
+                name: $name,
+                numLabelersRequired: 1,
+                questions: $questions,
+                sourceColumnId: $source_col_id,
+                instructions: ""
+            ) {
+                id
+            }
+        }
+    """
+
+    def __init__(
+        self, name: str, dataset_id: int, source_column_id: int, targets: List[str]
+    ):
+        questions = [
+            {"type": "ANNOTATION", "targets": targets, "keywords": [], "text": name,}
+        ]
+        super().__init__(
+            query=self.query,
+            variables={
+                "name": name,
+                "dataset_id": dataset_id,
+                "questions": questions,
+                "source_col_id": source_column_id,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            questionnaire = Questionnaire(
+                **super().process_response(response)["createQuestionnaire"]
+            )
+        except IndexError:
+            raise IndicoNotFound("Failed to create Questionnaire")
+        return questionnaire
+
+
+class CreateTeachTask(RequestChain):
```

Comment:
The name here should not be CreateTeachTask it's not consistent with our naming convention. If the above create questionnaire is only for internal request  it should just be _CreateQuestionnaire and this one CreateQuestionnaire 

Also this query pushes into the realm that I'd rather leave as an example then a task. It's implying a lot about how it would be used in context (from a csv, labels would be available for examples etc...) that is more the realm of a script than the python client.